Gems/Terrain/Code/Source/TerrainRenderer/TerrainFeatureProcessor.cpp

```diff
@@ -339,9 +362,47 @@ namespace Terrain
 
         uint16_t detailMaterialId = CreateOrUpdateDetailMaterial(material);
         materialRegion.m_materialsForSurfaces.push_back({ surfaceTag, detailMaterialId });
+        ++m_detailMaterials.GetData(detailMaterialId).refCount;
         m_dirtyDetailRegion.AddAabb(materialRegion.m_region);
     }
 
+    void TerrainFeatureProcessor::OnRenderPipelinePassesChanged([[maybe_unused]] AZ::RPI::RenderPipeline* renderPipeline)
+    {
+        CacheForwardPass();
+    }
+
+    void TerrainFeatureProcessor::CheckDetailMaterialForDeletion(uint16_t detailMaterialId)
+    {
+        auto& detailMaterialData = m_detailMaterials.GetData(detailMaterialId);
+        if (--detailMaterialData.refCount == 0)
+        {
+            uint16_t bufferIndex = detailMaterialData.m_detailMaterialBufferIndex;
+            DetailMaterialShaderData& shaderData = m_detailMaterialShaderData.GetElement(bufferIndex);
+
+            for (uint16_t imageIndex :
+                {
+                    shaderData.m_colorImageIndex,
+                    shaderData.m_normalImageIndex,
+                    shaderData.m_roughnessImageIndex,
+                    shaderData.m_metalnessImageIndex,
+                    shaderData.m_specularF0ImageIndex,
+                    shaderData.m_occlusionImageIndex,
+                    shaderData.m_heightImageIndex
+                })
+            {
+                if (imageIndex != InvalidDetailImageIndex)
+                {
+                    m_detailImageViews.at(imageIndex) = AZ::RPI::ImageSystemInterface::Get()->GetSystemImage(AZ::RPI::SystemImage::Magenta)->GetImageView();
```

Comment:
I know there were some recent changes around how Atom supports missing textures, should this get a special "missing texture" texture instead of magenta?

Reply:
I think this case is a little different. It's not really a missing texture, it's just a hole in the array, and _something_ needs to be bound there or it will complain. Nothing will actually reference this hole. I chatted to doug about this and it sounds like longer term we want null ImageViews in the array to be OK but that's just not the case right now so I'm binding a 1 pixel texture instead.